### PR TITLE
:sparkles: #11 - feat: add datagrid component

### PR DESCRIPTION
**Pull request by svenvandescheur**
Ready for review

### CODE REVIEW

File: src/components/datagrid/datagrid.scss [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    &__body {
+      display: block;
+    }
+
+    &__row {
+      background-color: var(
+        --typography-color-background
+      ); //border-radius: var(--border-radus-m);
```
Comment (Xaohs) [inline src/components/datagrid/datagrid.scss:77]: Can we remove this comment?

File: src/components/datagrid/datagrid.tsx [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+  /**
+   * Renders a cell based on type of `rowData[field]`.
+   * @param rowData
+   * @param field
+   * @param index
+   */
+  const renderCell = (rowData: RowData, field: string, index: number) => {
+    const fieldIndex = renderableFields.indexOf(field);
+    const urlField = urlFields.find((f) => String(rowData[f]).match(REGEX_URL));
```
Comment (Xaohs) [inline src/components/datagrid/datagrid.tsx:95]: Is it possible to make a re-usable function in libs for this?

File: src/components/datagrid/datagrid.tsx [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    // Implicit link: first column and `rowUrl` resolved using `urlFields`.
+    const isImplicitLink = rowUrl && fieldIndex === 0;
+
+    // Cell should be a link is truthy.
+    const link = isExplicitLink
+      ? String(data)
+      : isImplicitLink
+        ? String(rowUrl)
+        : "";
+
```
Comment (Xaohs) [inline src/components/datagrid/datagrid.tsx:112]: I understand why we are using the ternaries here, but it is very unreadable. Could we slightly refactor this for readability?

File: src/components/datagrid/datagrid.tsx [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+          </tr>
+        </thead>
+
+        {/* Cells */}
+        <tbody className="mykn-datagrid__body" role="rowgroup">
+          {results.map((rowData, index) => (
+            /**
+             * FIXME: This effectively still uses index as keys which might lead to issues.
+             * @see {@link https://react.dev/learn/rendering-lists#rules-of-keys|Rules of keys}
+             */
```
Comment (Xaohs) [inline src/components/datagrid/datagrid.tsx:193]: Could we apply the FIXME? :smile:

File: src/components/datagrid/datagrid.tsx [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+      </table>
+    </div>
+  );
+};
+
+/**
+ * Converts "field_name" to "FIELD NAME".
+ * @param field
+ */
+const field2Caption = (field: string): string =>
```
Comment (Xaohs) [inline src/components/datagrid/datagrid.tsx:226]: Should we make a function in lib for this?

**Review by Xaohs — CHANGES_REQUESTED** [review]
I have done some code checks, visually: 

- We are missing checkboxes to select each datagrid, and the button to do something with the selected ones. Are we leaving this aside for now? 
- The table headers are not aligned correctly according to the design. 
- Could you take another look at the mobile design, the aligning of the values seems a bit unnatural to me. 
- Upon closer inspection, the borders on the pagination buttons are always lit up (blueish), but in the design they should be more greyed out. Any thoughts on this?

File: src/components/datagrid/datagrid.tsx (hunk 4 above)
Comment (svenvandescheur) [reply src/components/datagrid/datagrid.tsx:193]: Will do so in a future PR containing the sorting functionality.

File: src/components/datagrid/datagrid.tsx (hunk 2 above)
Comment (svenvandescheur) [reply src/components/datagrid/datagrid.tsx:95]: I moved field2caption to libs, extracting the urlField seems to be very specific logic to me. I don't think we should move that to libs for now.

**Comment by svenvandescheur** [comment]
> I have done some code checks, visually:
> 
> * We are missing checkboxes to select each datagrid, and the button to do something with the selected ones. Are we leaving this aside for now?

Yes, they might be added in the future but we postponed that feature for now.

> * The table headers are not aligned correctly according to the design.

- [ ] Fix the design alignment and sizing of columns as close as possible.

> * Could you take another look at the mobile design, the aligning of the values seems a bit unnatural to me.

- [ ] Attempt to improve mobile design

> * Upon closer inspection, the borders on the pagination buttons are always lit up (blueish), but in the design they should be more greyed out. Any thoughts on this?

- [ ] Inspect color usage of pagination

- [ ] Rebase main

**Comment by svenvandescheur** [comment]
@Xaohs merging to unblock, additional review is still appreciated.

**Pull request merged**